components/support/webextensions/src/main/java/mozilla/components/support/webextensions/WebExtensionSupport.kt

```diff
@@ -176,6 +227,66 @@ object WebExtensionSupport {
             }
     }
 
+    /**
+     * Indicates the status of a request for updating an extension.
+     */
+    internal sealed class ExtensionUpdateStatus {
+        /**
+         * The extension is not part of the installed list.
+         */
+        object NotInstalled : ExtensionUpdateStatus()
+        /**
+         * The extension was successfully updated.
+         */
+        object SuccessfullyUpdated : ExtensionUpdateStatus()
+        /**
+         * The extension an error happens while trying to update.
+         */
+        class Error(val message: String, val exception: Throwable) : ExtensionUpdateStatus()
+    }
+
+    /**
+     * Updates the provided [extensionId] by updating it in the the [store] and re-attaching
+     * all the handlers.
+     */
+    internal fun updateInstalledExtension(extensionId: String, onFinish : ((ExtensionUpdateStatus) -> Unit)) {
+        val extension = installedExtensions[extensionId]
+
+        if (extension == null) {
+            onFinish(ExtensionUpdateStatus.NotInstalled)
```

Comment:
Curious how do you see this happen? The user uninstalled the extension and the background worker to update is still running?

Reply:
That maybe the situation, I'm adding api for `unregisterForFutureUpdates` to avoid that.